src/SablierV2MerkleLockupLT.sol

```diff
@@ -0,0 +1,160 @@
+        // preimage attacks.
+        bytes32 leaf = keccak256(bytes.concat(keccak256(abi.encode(index, recipient, amount))));
+
+        // Checks: validate the function.
+        _checkClaim(index, leaf, merkleProof);
+
+        // Calculate the tranches based on the `amount`.
+        LockupTranched.TrancheWithDuration[] memory tranches = _calculateTranches(amount);
+
+        // Effects: mark the index as claimed.
+        _claimedBitMap.set(index);
+
+        // Interactions: create the stream via {SablierV2LockupTranched}.
+        streamId = LOCKUP_TRANCHED.createWithDurations(
+            LockupTranched.CreateWithDurations({
+                sender: admin,
+                recipient: recipient,
+                totalAmount: amount,
+                asset: ASSET,
+                cancelable: CANCELABLE,
+                transferable: TRANSFERABLE,
+                tranches: tranches,
+                broker: Broker({ account: address(0), fee: ud(0) })
+            })
+        );
+
+        // Log the claim.
+        emit Claim(index, recipient, amount, streamId);
+    }
+
+    /// @dev Calculates the stream tranches based on Merkle tree amount and unlock percentage for each tranche.
+    function _calculateTranches(uint128 amount)
+        internal
+        view
+        returns (LockupTranched.TrancheWithDuration[] memory tranches)
+    {
+        // Load the tranches in memory to save gas.
+        MerkleLockupLT.TrancheWithPercentage[] memory tranchesWithPercentages = _tranchesWithPercentages;
+
+        // Declare the variables need for calculation.
+        UD60x18 trancheAmountsSum;
+        uint256 trancheCount = tranchesWithPercentages.length;
+        tranches = new LockupTranched.TrancheWithDuration[](trancheCount);
+
+        UD60x18 udAmount = ud(amount);
+
+        // Iterate over each tranche to calculate its amount based on its percentage.
+        for (uint256 i = 0; i < trancheCount; ++i) {
+            // Convert the tranche's percentage to `UD60x18` for calculation.
+            UD60x18 percentage = (tranchesWithPercentages[i].unlockPercentage).intoUD60x18();
+
+            // Calculate the tranche's amount by applying its percentage to the `amount`.
+            UD60x18 trancheAmount = udAmount.mul(percentage);
+
+            // Sum all tranche amounts.
+            trancheAmountsSum = trancheAmountsSum.add(trancheAmount);
+
+            // Assign calculated amount and duration to the tranche.
+            tranches[i] = LockupTranched.TrancheWithDuration({
+                amount: trancheAmount.intoUint128(),
+                duration: tranchesWithPercentages[i].duration
+            });
+        }
+
+        // Adjust the last tranche amount to prevent claim failure due to rounding differences during calculations. We
+        // need to ensure the protocol invariant: the sum of all tranches' amounts equals the deposit amount.
```

Comment:
since we are going to remove the protocol fee, it is now correct to use "claim" instead of "deposit", before it wasn't because `claim > deposit`

Reply:
I think even back then, calling it deposit amount doesn't look apt. In the presence of protocol fee, this would have been `claimAmount - protocolFee` which is still not the deposit amount.

Also, in your last commit, I don't see this change.